app/services/course/assessment/programming_codaveri_evaluation_service.rb

```diff
@@ -137,7 +137,7 @@ def construct_grading_object
     @answer_object[:problemId] = @question.codaveri_id
 
     @answer_object[:languageVersion][:language] = @question.language.polyglot_name
-    @answer_object[:languageVersion][:version] = @question.language.polyglot_version
+    @answer_object[:languageVersion][:version] = @question.language.extend(CodaveriLanguageConcern).polyglot_version
 
     @answer.files.each do |file|
       file_template = default_codaveri_student_file_template
```

Comment:
if this is a code-submission, don't we also need to extract the filename from the public class?

Reply:
done